ivy/functional/ivy/activations.py

```diff
@@ -506,17 +553,25 @@ def softmax(
 @to_native_arrays_and_back
 @handle_array_function
 @handle_device_shifting
+@handle_complex_input
 def softplus(
     x: Union[ivy.Array, ivy.NativeArray],
     /,
     *,
     beta: Optional[Union[int, float]] = None,
     threshold: Optional[Union[int, float]] = None,
     out: Optional[ivy.Array] = None,
+    complex_mode: Literal["split", "magnitude", "jax"] = "jax",
 ) -> ivy.Array:
     """
     Apply the softplus function element-wise.
 
+    If the input is complex, then by default we apply the softplus operation
+    `log(1+ exp(x))` to  each element
+    If threshold is set we check if either its real part is strictly negative or
+    if its real part is zero and its imaginary part is negative then we apply
+    `input×β > threshold`.
+
```

Comment:
I find this docstring quite confusing, could you reword it a bit? It still seems to reference the number being negative.

Although, I suppose instead you could remove this and edit the description of the `threshold` parameter (since this just describes the standard way softplus works, and then describes special behaviour of threshold with complex inputs)